drivers/hid/usbhid/hid-core.c

```diff
@@ -1388,17 +1388,21 @@ static int usbhid_probe(struct usb_interface *intf, const struct usb_device_id *
 	else if (intf->cur_altsetting->desc.bInterfaceProtocol == 0)
 		hid->type = HID_TYPE_USBNONE;
 
-	if (dev->manufacturer)
-		strscpy(hid->name, dev->manufacturer, sizeof(hid->name));
+	snprintf(hid->name, sizeof(hid->name), "usb-%s ", dev_name(&dev->dev));
+
+	if (dev->manufacturer) {
+		strlcat(hid->name, dev->manufacturer, sizeof(hid->name));
+	}
 
 	if (dev->product) {
 		if (dev->manufacturer)
 			strlcat(hid->name, " ", sizeof(hid->name));
 		strlcat(hid->name, dev->product, sizeof(hid->name));
 	}
 
-	if (!strlen(hid->name))
-		snprintf(hid->name, sizeof(hid->name), "HID %04x:%04x",
+	if (strlen(hid->name) == strlen(dev_name(&dev->dev)))
```

Comment:
Given the snprintf above, and the fact that other changes just append, how can this condition ever be true?

Reply:
It can't.
Too many revisions switching from strscpy to snprintf (to add the usb- prefix), and missing updating this condition.

Ideally this issue wants to be sorted further up the stack (xinput was always happy with multiple identical devices, although you had to reference them by id number rather than name), so I'll have a word with Simon and David T later on.